### ISSUE TITLE
Opening Coq .v files really slow

### ISSUE THREAD

**Issue by bn-peters** [doomemacs/doomemacs#5823]
### What did you expect to happen?

Opening a .v file (e.g. []) takes a long time.

### What actually happened?

.

### Describe your attempts to resolve the issue

Profiling results in the following backtrace:
```
- ...                                                                                         22058  94%
 - doom--fix-broken-smie-modes-a                                                              21094  90%
  - let                                                                                       21094  90%
   - let                                                                                      21094  90%
    - let*                                                                                    21094  90%
     - unwind-protect                                                                         21094  90%
      - progn                                                                                 21094  90%
       - let                                                                                  21094  90%
        - let*                                                                                21094  90%
         - unwind-protect                                                                     21094  90%
          - progn                                                                             21094  90%
           - funcall                                                                          21094  90%
            - #<compiled 0x155ea2777f99>                                                      21094  90%
             - smie-config-guess                                                              21093  90%
              - condition-case                                                                21093  90%
               - funcall                                                                      21093  90%
                - #<compiled 0x155ea2a248e5>                                                  21093  90%
                 - smie-config--guess                                                         21092  90%
                  - smie-config--get-trace                                                    21088  90%
                   + smie-indent-calculate                                                    21079  90%
                   - advice--add-function                                                         6   0%
                    + #<compiled 0x155ea1f2210d>                                                  2   0%
                      #<compiled 0x155ea2780e2d>                                                  1   0%
                      advice--make                                                                1   0%
                     #<compiled 0x155ea240d3cd>                                                   1   0%
             + dtrt-indent-try-set-offset                                                         1   0%
   Automatic GC                                                                                 951   4%
 + flycheck-mode-on-safe                                                                          6   0%
 + flycheck-buffer                                                                                6   0%
 + vc-refresh-state                                                                               1   0%
+ command-execute                                                                              1216   5%
+ redisplay_internal (C function)                                                                 2   0%
+ gcmh-register-idle-gc                                                                           1   0%
+ #<compiled 0x155ea29b2ee9>                                                                      1   0%
+ timer-event-handler                                                                             1   0%
+ sp--save-pre-command-state                                                                      1   0%
```

Under some other situations I also had dtrt cause the slow-down, but it was always related to SMIE.

### Steps to reproduce

1. Open Emacs with coq plugin enabled
2. Download and open a large Coq file, e.g. https://gitlab.mpi-sws.org/FP/semantics-course/-/raw/main/theories/systemf/binary_logrel.v

### System Information

https://pastebin.com/Tn27gfbe

**Comment by mpardalos** [comment]
I have had this issue as well, I got this fix from a colleague, so I am not quite sure why it works, but it did the trick for me.

```elisp
;; Fix for slow startup
(after! core-editor
  (add-to-list 'doom-detect-indentation-excluded-modes 'coq-mode))
```

**Comment by bn-peters** [comment]
This works as a workaround; Thank you!

**Comment by mpardalos** [comment]
Would this be a reasonable fix to add into the module? Does anyone have any ideas as to why this works?

**Comment by axelson** [comment]
FYI `after! core-editor` is no longer being triggered for me so I've adjusted the snippet to 

```elisp
(after! coq-mode
  (add-to-list 'doom-detect-indentation-excluded-modes 'coq-mode))
```
(although I'm actually making the change for `elixir-mode`)

There's probably a better way of doing this but I couldn't find a replacement for `after! core-editor`. The downside of this approach is if the very first coq-mode file you load is large then it'll still be slow.

**Comment by hlissner** [comment]
Just an FYI, `core-editor` was moved to `doom-editor` some time ago, that's why `(after! core-editor ...)` no longer works, *but* that `after!` block isn't even necessary in the first place (for any `doom-*` feature), so `(add-to-list 'doom-detect-indentation-excluded-modes 'coq-mode)` is enough.

That said, I've added `coq-mode` to `doom-detect-indentation-excluded-modes` in d03ac05, which should resolve this issue for *.v files, at the very least.
